frontend/packages/erd-core/src/features/erd/components/ERDContent/components/TableNode/TableNode.tsx

```diff
@@ -21,6 +21,52 @@ export const TableNode: FC<Props> = ({ data }) => {
   const showMode = data.showMode ?? _showMode
   const name = data?.table?.name
 
+  const textRef = useRef<HTMLSpanElement>(null)
+  const [isTruncated, setIsTruncated] = useState<boolean>(false)
+
+  useEffect(() => {
+    const checkTruncation = () => {
+      if (!textRef.current) return
+
+      const element = textRef.current
+
+      const style = window.getComputedStyle(element)
+
+      // Create a canvas to measure text width
+      const canvas = document.createElement('canvas')
+      const context = canvas.getContext('2d')
+      if (!context) return
+
+      // Set the font to match the element
+      context.font = style.font
+
+      // Measure the text width
+      const textMetrics = context.measureText(name)
+      const textWidth = textMetrics.width
```

Comment:
The purpose of using `canvas.getContext('2d').measureText()` in this implementation is, I understand, to measure the width per element “without depending on the layout and without causing forced reflow” so that even ERD with many nodes can maintain performance when scrolling and pan/zooming. I understood that the intention was to maintain performance when scrolling and panning/zooming, even with ERDs with many nodes.